### PR TITLE
[SPARK-35696][PYTHON][DOCS] Refine the code examples in pandas-on-Spark documentation

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to refine the code examples for pandas-on-Spark since some of them still follows the naming for Koalas.

For example,

```python
kdf = ks.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
```

should be refined to

```python
psdf = ps.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
```

Also fixed the several remaining Koalas stuffs in FAQ

### Why are the changes needed?

Because we don't want to use the name "Koalas" in the Apache Spark anymore.

### Does this PR introduce _any_ user-facing change?

Yes, the examples in the documentation will be changed with refined names.

### How was this patch tested?

Manually built the docs and check one by one.

### CODE REVIEW

File: python/docs/source/getting_started/ps_10mins.ipynb [hunk 1]
```diff
@@ -44,7 +44,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "s = ks.Series([1, 3, 5, np.nan, 6, 8])"
+    "s = ps.Series([1, 3, 5, np.nan, 6, 8])"
```
Comment (HyukjinKwon) [inline python/docs/source/getting_started/ps_10mins.ipynb:47]: @itholic, can you revert the changes in ipynb files? I will fix it in https://github.com/apache/spark/pull/33041

Comment (itholic) [reply python/docs/source/getting_started/ps_10mins.ipynb]: Sure, just revert the changes

File: python/docs/source/user_guide/pandas_on_spark/faq.rst [hunk 2]
```diff
@@ -2,30 +2,25 @@
 FAQ
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/faq.rst:2]: Nice, thanks for the updates here!

File: python/docs/source/user_guide/pandas_on_spark/best_practices.rst [hunk 3]
```diff
@@ -54,9 +54,9 @@ before the actual computation since pandas API on Spark is based on lazy executi
 .. code-block:: python
 
    >>> import pyspark.pandas as ks
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/best_practices.rst:56]:
```suggestion
   >>> import pyspark.pandas as ps
```

File: python/docs/source/user_guide/pandas_on_spark/options.rst [hunk 4]
```diff
@@ -78,7 +78,7 @@ are restored automatically when you exit the `with` block:
 
 .. code-block:: python
 
-   >>> with ks.option_context("display.max_rows", 10, "compute.max_rows", 5):
+   >>> with ps.option_context("display.max_rows", 10, "compute.max_rows", 5):
    ...    print(ks.get_option("display.max_rows"))
    ...    print(ks.get_option("compute.max_rows"))
```
Comment (ueshin) [inline python/docs/source/user_guide/pandas_on_spark/options.rst:83]: There are still `ks` here and lines 86 and 87.

Comment (itholic) [reply python/docs/source/user_guide/pandas_on_spark/options.rst:83]: Thanks!!

**Comment by HyukjinKwon** [comment]
@itholic can you bring `python/docs/source/getting_started/quickstart_ps.ipynb ` back?

**Comment by itholic** [comment]
> @itholic can you bring `python/docs/source/getting_started/quickstart_ps.ipynb ` back?

It was mistakenly removed. Just revert!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Related tests passed. The test failure is fixed in https://github.com/apache/spark/commit/37f70422b54f50aecf335177ca791a4c2ec51555.

Merged to master.

File: python/docs/source/user_guide/pandas_on_spark/faq.rst [hunk 5]
```diff
@@ -2,30 +2,25 @@
...
 
-Should I use PySpark's DataFrame API or Koalas?
------------------------------------------------
-
-If you are already familiar with pandas and want to leverage Spark for big data, we recommend
-using Koalas. If you are learning Spark from ground up, we recommend you start with PySpark's API.
-
-Does Koalas support Structured Streaming?
+Does pandas API on Spark support Structured Streaming?
 -----------------------------------------
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/faq.rst:12]: @itholic can you fix this underbars? it causes warnings:

```
/.../spark/python/docs/source/user_guide/pandas_on_spark/faq.rst:12: WARNING: Title underline too short.

Does pandas API on Spark support Structured Streaming?
-----------------------------------------
/.../spark/python/docs/source/user_guide/pandas_on_spark/faq.rst:12: WARNING: Title underline too short.

Does pandas API on Spark support Structured Streaming?
-----------------------------------------
```

Comment (HyukjinKwon) [reply python/docs/source/user_guide/pandas_on_spark/faq.rst:12]: I don't know how the CI passed since we turn warnings as errors.

Comment (itholic) [reply python/docs/source/user_guide/pandas_on_spark/faq.rst:12]: Yes, I'll fix it in the follow-up

**Pull request closed without merging**